lib/elixir/lib/kernel.ex

```diff
@@ -2606,6 +2642,9 @@ defmodule Kernel do
       true
 
   """
+  # TODO: FIX: following line gives error
+  # @spec (first .. last) :: Range.t(first, last) when first: integer, last: integer
+  @spec ((Macro.t | any)..(Macro.t | any)) :: Range.t | no_return
```

Comment:
`..` in specs accept integer. I wouldn't spec this function because it is confusing to think about it receiving expressions.